src/magnet_uri.cpp

```diff
@@ -190,7 +190,8 @@ namespace libtorrent {
 				error_code e;
 				display_name = unescape_string(value, e);
 			}
-			else if (name == "tr"_sv) // tracker
+			else if (name == "tr"_sv ||
+			    (name.substr(0, 3) == "tr."_sv && is_numeric(name.substr(3, name.size())))) // tracker
```

Comment:
surely the number following the `tr.` name has some significance, no? Does it not specify which tier a tracker belongs to? If it doesn't, what's the point?

Can you point to any documentation of what this is supposed to mean?

Reply:
This would need a test as well, please add another `TORRENT_TEST()` to `test/test_magnet.cpp` to ensure that trackers are parsed out of magnets using this syntax correctly.

There is a large test case in there that really should be split up into smaller ones. Don't use that as inspiration, write a small test that just tests all aspects of this one feature.